pkgs/by-name/bp/bpftop/package.nix

```diff
@@ -28,6 +31,7 @@ rustPlatform.buildRustPackage rec {
 
   nativeBuildInputs = [
     pkg-config
+    clang
```

Comment:
Does this specifically need clang? Or will `stdenv.cc` suffice? If it specifically needs clang, please put a comment why,

Reply:
Or we should overwrite buildRustPackage to use the clangStdenv?

Reply:
Yes, it doesn't compile otherwise.
```
bpftop> error: failed to run custom build command for `bpftop v0.5.1 (/build/source)`
bpftop>
bpftop> Caused by:
bpftop>   process didn't exit successfully: `/build/source/target/release/build/bpftop-62a96e26a5815cd3/build-script-build` (exit status: 101)
bpftop>   --- stderr
bpftop>   thread 'main' panicked at build.rs:21:10:
bpftop>   called `Result::unwrap()` on an `Err` value: failed to build `src/bpf/pid_iter.bpf.c`
bpftop>
bpftop>   Caused by:
bpftop>       0: Failed to execute clang
bpftop>       1: No such file or directory (os error 2)
bpftop>   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
bpftop> warning: build failed, waiting for other jobs to finish...
```

with `RUST_BACKTRACE= 1;`
```
bpftop> error: failed to run custom build command for `bpftop v0.5.1 (/build/source)`
bpftop> note: To improve backtraces for build dependencies, set the CARGO_PROFILE_RELEASE_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.
bpftop>
bpftop> Caused by:
bpftop>   process didn't exit successfully: `/build/source/target/release/build/bpftop-62a96e26a5815cd3/build-script-build` (exit status: 101)
bpftop>   --- stderr
bpftop>   thread 'main' panicked at build.rs:21:10:
bpftop>   called `Result::unwrap()` on an `Err` value: failed to build `src/bpf/pid_iter.bpf.c`
bpftop>
bpftop>   Caused by:
bpftop>       0: Failed to execute clang
bpftop>       1: No such file or directory (os error 2)
bpftop>
bpftop>   Stack backtrace:
bpftop>      0: <E as anyhow::context::ext::StdError>::ext_context
bpftop>      1: anyhow::context::<impl anyhow::Context<T,E> for core::result::Result<T,E>>::context
bpftop>      2: libbpf_cargo::build::check_clang
bpftop>      3: libbpf_cargo::build::build_single
bpftop>      4: libbpf_cargo::SkeletonBuilder::build
bpftop>      5: libbpf_cargo::SkeletonBuilder::build_and_generate
bpftop>      6: build_script_build::main
bpftop>      7: core::ops::function::FnOnce::call_once
bpftop>      8: std::sys_common::backtrace::__rust_begin_short_backtrace
bpftop>      9: std::rt::lang_start::{{closure}}
bpftop>     10: std::panicking::try
bpftop>     11: std::rt::lang_start_internal
bpftop>     12: std::rt::lang_start
bpftop>     13: main
bpftop>     14: __libc_start_call_main
bpftop>     15: __libc_start_main@@GLIBC_2.34
bpftop>     16: _start
bpftop>   stack backtrace:
bpftop>      0: rust_begin_unwind
bpftop>      1: core::panicking::panic_fmt
bpftop>      2: core::result::unwrap_failed
bpftop>      3: build_script_build::main
bpftop>      4: core::ops::function::FnOnce::call_once
bpftop>   note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
bpftop> warning: build failed, waiting for other jobs to finish...
```

with `RUST_BACKTRACE= "full";`
```
bpftop>    Compiling tracing-journald v0.3.0
bpftop> error: failed to run custom build command for `bpftop v0.5.1 (/build/source)`
bpftop> note: To improve backtraces for build dependencies, set the CARGO_PROFILE_RELEASE_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.
bpftop>
bpftop> Caused by:
bpftop>   process didn't exit successfully: `/build/source/target/release/build/bpftop-62a96e26a5815cd3/build-script-build` (exit status: 101)
bpftop>   --- stderr
bpftop>   thread 'main' panicked at build.rs:21:10:
bpftop>   called `Result::unwrap()` on an `Err` value: failed to build `src/bpf/pid_iter.bpf.c`
bpftop>
bpftop>   Caused by:
bpftop>       0: Failed to execute clang
bpftop>       1: No such file or directory (os error 2)
bpftop>
bpftop>   Stack backtrace:
bpftop>      0: <E as anyhow::context::ext::StdError>::ext_context
bpftop>      1: anyhow::context::<impl anyhow::Context<T,E> for core::result::Result<T,E>>::context
bpftop>      2: libbpf_cargo::build::check_clang
bpftop>      3: libbpf_cargo::build::build_single
bpftop>      4: libbpf_cargo::SkeletonBuilder::build
bpftop>      5: libbpf_cargo::SkeletonBuilder::build_and_generate
bpftop>      6: build_script_build::main
bpftop>      7: core::ops::function::FnOnce::call_once
bpftop>      8: std::sys_common::backtrace::__rust_begin_short_backtrace
bpftop>      9: std::rt::lang_start::{{closure}}
bpftop>     10: std::panicking::try
bpftop>     11: std::rt::lang_start_internal
bpftop>     12: std::rt::lang_start
bpftop>     13: main
bpftop>     14: __libc_start_call_main
bpftop>     15: __libc_start_main@@GLIBC_2.34
bpftop>     16: _start
bpftop>   stack backtrace:
bpftop>      0:     0x555555737d47 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h4886110ae7835fd2
bpftop>      1:     0x555555766740 - core::fmt::write::h6fab83f0f5bc0c8f
bpftop>      2:     0x5555557518b5 - std::io::Write::write_fmt::h64824cb874054607
bpftop>      3:     0x555555737b14 - std::sys_common::backtrace::print::hb8fea9d041077138
bpftop>      4:     0x555555738687 - std::panicking::default_hook::{{closure}}::hc60f0f690c67943c
bpftop>      5:     0x5555557383e7 - std::panicking::default_hook::h56281bcc5d6e2f17
bpftop>      6:     0x555555738bf8 - std::panicking::rust_panic_with_hook::hd61e30fe5971d187
bpftop>      7:     0x555555738ada - std::panicking::begin_panic_handler::{{closure}}::h95d2684e28f89fd2
bpftop>      8:     0x555555737f76 - std::sys_common::backtrace::__rust_end_short_backtrace::h48780d898316fff8
bpftop>      9:     0x555555738824 - rust_begin_unwind
bpftop>     10:     0x5555555975a5 - core::panicking::panic_fmt::hdb89b5c14a056cec
bpftop>     11:     0x555555597a43 - core::result::unwrap_failed::h850510ea6c159887
bpftop>     12:     0x555555599d61 - build_script_build::main::hf43c62a60e22f51d
bpftop>     13:     0x555555598c23 - core::ops::function::FnOnce::call_once::he00d7bb0ca569e9f
bpftop>     14:     0x555555598836 - std::sys_common::backtrace::__rust_begin_short_backtrace::h632c3a247265fce5
bpftop>     15:     0x555555598889 - std::rt::lang_start::{{closure}}::h244ef22e5ea1ac3a
bpftop>     16:     0x555555738724 - std::panicking::try::ha86251cf5daa9bea
bpftop>     17:     0x55555572946b - std::rt::lang_start_internal::ha6a51778162f8d22
bpftop>     18:     0x555555598867 - std::rt::lang_start::h6da3b6347a33d464
bpftop>     19:     0x555555599e95 - main
bpftop>     20:     0x7ffff7d9e10e - __libc_start_call_main
bpftop>     21:     0x7ffff7d9e1c9 - __libc_start_main@@GLIBC_2.34
bpftop>     22:     0x555555597e55 - _start
bpftop>     23:                0x0 - <unknown>
bpftop> warning: build failed, waiting for other jobs to finish...
```

Reply:
try clangStdenv